jsonrpc/params.go

```diff
@@ -92,6 +98,22 @@ func (p Params) UnmarshalInto(receivers ...interface{}) error {
 		return errors.New("not enough params to decode")
 	}
 
+	// Return an array of the receivers' types
+	receiversType := listTypes(receivers)
+
+	var paramElement []string
+	for _, i := range p {
+		paramElement = append(paramElement, string(i))
+	}
+
+	// Return p Params in json.Rawmessage type with [] to be parsed
+	rawParams := json.RawMessage("[" + strings.Join(paramElement, ",") + "]")
+
+	_, err := parsePositionalArguments(rawParams, receiversType)
+	if err != nil {
+		return err
+	}
+
 	for i, r := range receivers {
```

Comment:
Since you already collected the reflected values of your params you can use them to set the values to the receivers
```
receiversValues, err := parsePositionalArguments(rawParams, receiversType)
if err != nil {
     return err
}
 for i, r := range receivers {
        if receiversValues[i].IsZero() {
            continue
        }
        reflect.ValueOf(r).Elem().Set(receiversValues[i].Elem())
    }
```

Reply:
changed it